src/components/createBook/CreateBook.tsx

```diff
@@ -1,26 +1,167 @@
+
+import { ChangeEvent, useEffect, useState } from 'react'
+
+import { MdAddCircleOutline as AddIcon } from 'react-icons/md'
+import { useNavigate } from 'react-router-dom'
+import Select, { MultiValue } from 'react-select'
+
 import noCover from '../../assets/no-cover.png'
+import AuthorResponse from '../../interfaces/AuthorResponse'
+import Book from '../../interfaces/Book'
+import { getAuthors } from '../../services/author.service'
+import { create } from '../../services/book.service'
+import CreateAuthor from '../createAuthor/CreateAuthor'
 import './CreateBook.css'
 
+const initialBook: Book = { Title: '', ISBN: '', Quantity: 0, AuthorsIds: [], Description: '', PublishDate: '' }
+
 function CreateBook() {
+  const [ authorList, setAuthorList ] = useState<AuthorResponse[]>([])
+  const [ selectedAuthors, setSelectedAuthors ] = useState<AuthorResponse[]>([])
+  const [ isAuthorsChanged, setIsAuthorsChanged ] = useState(false)
+  const [ cover, setCover ] = useState(noCover)
+  const [ show, setShow ] = useState(false)
+  const [ titleErrorMessage, setTitleErrorMessage ] = useState ('')
+  const [ ISBNErrorMessage, setISBNErrorMessage ] = useState ('')
+  const [ quantityErrorMessage, setQuantityErrorMessage ] = useState ('')
+  const [ requestCover, setRequestCover ] = useState<Blob>(new Blob)
+  const [ book, setBook ] = useState<Book>(initialBook)
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    fetchAuthors()
+  }, [ isAuthorsChanged ])
+
+  const fetchAuthors = () => {
+    getAuthors().then(response => {
+      setAuthorList(response.data)
+    }).catch(error => {console.error(error)})
+  }
+
+  const handleImageChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const files = event.target?.files
+    const reader = new FileReader()
+    if(files!= null) {
+      reader.readAsDataURL(files[0])
+      setRequestCover(files[0])
+      reader.onloadend = function () {
+        const base64data = reader.result
+        if(base64data)
+          setCover(base64data as string)
+      }
+    }
+  }
+
+  const handleSelectedAuthorsChange = (authorsData: MultiValue<AuthorResponse>) => {
+    setSelectedAuthors(authorsData as AuthorResponse[])
+  }
+
+  const createBook = () => {
+    if(validateInput()) {
+      create(prepareFormData())
```

Comment:
In this case, you should know what service is being used (create what ? ) - hint: don't export function by function from service, just implement functions without export keyword and at the end of the file use default export to wrap all the functions in one object and export them.
Or use more descriptive names at the service, but in that case you will have to import function by function